### ISSUE TITLE
rserver has unreasonably short limits on filepath lengths for configuration

### ISSUE THREAD

**Issue by gdevenyi** [rstudio/rstudio#15024]
### System details

    RStudio Edition : Server
    RStudio Version : 2024.08.0-342
    OS Version      : Ubuntu 22.04
    R Version       : 4.1.3

### Steps to reproduce the problem

Execute rserver with the following command
```
/opt/quarantine/software/rstudio-server/2024.08.0-342/install/bin/rserver \
--server-working-dir /home/cic/devgab \
--auth-none=1 --www-frame-origin=same \
--www-port=63726 --www-address=0.0.0.0 \
--www-verify-user-agent=0 \
--server-data-dir=/home/cic/devgab/.local/rstudio-server/56d57242-2995-44b9-bbbd-cdc2a7db148b \
--server-pid-file=/home/cic/devgab/.local/rstudio-server/56d57242-2995-44b9-bbbd-cdc2a7db148b/rstudio.pid \
--server-user=devgab \
--database-config-file=/home/cic/devgab/.local/rstudio-server/56d57242-2995-44b9-bbbd-cdc2a7db148b/database.conf
```

Contents of `database.conf`
```
directory=/home/cic/devgab/.local/rstudio-server/56d57242-2995-44b9-bbbd-cdc2a7db148b
```

### Describe the problem in detail
rserver bails out with error:
```
TTY detected. Printing informational message about logging configuration. Logging configuration loaded from '/etc/rstudio/logging.conf'. Logging to '/home/cic/devgab/.local/share/rstudio/log/rserver.log'.
2024-07-31T16:26:50.011828Z [rserver] ERROR Unexpected exception: File name too long [system:36]; LOGGED FROM: int main(int, char* const*) src/cpp/server/ServerMain.cpp:975
```

### Describe the behavior you expected
Runs normally.

I have tried this without the extra 56d57242-2995-44b9-bbbd-cdc2a7db148b component to shorten the path and the exact same commands work.

I believe one of the options/variables (can't tell which one from the error) has an unreasonably short string buffer.

- [X] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [X] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [NA] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.posit.co/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by gtritchie** [comment]
I was able to repro via setting `server-data-dir=/home/gary/.local/rstudio-server/more-stuff-here/56d57242-2995-44b9-bbbd-cdc2a7db148b`.

**Comment by kevinushey** [comment]
For reference, the issue occurs because we try to create a domain socket within server-data-dir. Here's a backtrace (testing on macOS, but the underlying cause should be the same)

```
(lldb) bt
* thread #1, queue = 'com.apple.main-thread', stop reason = breakpoint 1.1
  * frame #0: 0x000000019ca34158 libc++abi.dylib`__cxa_throw
    frame #1: 0x0000000100ae21ac rserver`void rstudio_boost::throw_exception<rstudio_boost::system::system_error>(e=0x000000016f379258, loc=0x000000016f3792d8) at throw_exception.hpp:171:5
    frame #2: 0x0000000100aeaa44 rserver`rstudio_boost::asio::detail::do_throw_error(err=0x000000016f3792f0, loc=0x000000016f3792d8) at throw_error.ipp:33:3
    frame #3: 0x0000000100aea968 rserver`rstudio_boost::asio::detail::throw_error(err=0x000000016f3792f0, loc=0x000000016f3792d8) at throw_error.hpp:42:5
    frame #4: 0x0000000100aea898 rserver`rstudio_boost::asio::local::detail::endpoint::init(this=0x000000016f379580, path_name="/Users/kevin/.local/rstudio-server/more-stuff-here/56d57242-2995-44b9-bbbd-cdc2a7db148b/rstudio-rserver/session-server-rpc.socket", path_length=129) at endpoint.ipp:113:5
    frame #5: 0x0000000100bd6ac0 rserver`rstudio_boost::asio::local::detail::endpoint::endpoint(this=0x000000016f379580, path_name="/Users/kevin/.local/rstudio-server/more-stuff-here/56d57242-2995-44b9-bbbd-cdc2a7db148b/rstudio-rserver/session-server-rpc.socket") at endpoint.ipp:49:3
    frame #6: 0x0000000100bd6a70 rserver`rstudio_boost::asio::local::detail::endpoint::endpoint(this=0x000000016f379580, path_name="/Users/kevin/.local/rstudio-server/more-stuff-here/56d57242-2995-44b9-bbbd-cdc2a7db148b/rstudio-rserver/session-server-rpc.socket") at endpoint.ipp:48:1
    frame #7: 0x0000000100bd6a3c rserver`rstudio_boost::asio::local::basic_endpoint<rstudio_boost::asio::local::stream_protocol>::basic_endpoint(this=0x000000016f379580, path_name="/Users/kevin/.local/rstudio-server/more-stuff-here/56d57242-2995-44b9-bbbd-cdc2a7db148b/rstudio-rserver/session-server-rpc.socket") at basic_endpoint.hpp:76:7
    frame #8: 0x0000000100bd68a8 rserver`rstudio_boost::asio::local::basic_endpoint<rstudio_boost::asio::local::stream_protocol>::basic_endpoint(this=0x000000016f379580, path_name="/Users/kevin/.local/rstudio-server/more-stuff-here/56d57242-2995-44b9-bbbd-cdc2a7db148b/rstudio-rserver/session-server-rpc.socket") at basic_endpoint.hpp:77:3
    frame #9: 0x0000000100bd6394 rserver`rstudio::core::http::initLocalStreamAcceptor(acceptorService=0x0000000137e08238, localStreamPath=0x0000000137e08528, fileMode=ALL_READ_WRITE) at LocalStreamSocketUtils.hpp:54:30
    frame #10: 0x0000000100adbd00 rserver`rstudio::core::http::LocalStreamAsyncServer::init(this=0x0000000137e081f0, localStreamPath=0x000000016f3797f0) at LocalStreamAsyncServer.hpp:86:14
    frame #11: 0x0000000100d16974 rserver`rstudio::server::session_rpc::initialize() at ServerSessionRpc.cpp:235:97
    frame #12: 0x0000000100a8f590 rserver`main(argc=5, argv=0x000000016f37ad88) at ServerMain.cpp:852:15
    frame #13: 0x000000019c6f20e0 dyld`start + 2360
```

If I understand correctly, the path length is checked here by Boost:

https://github.com/boostorg/asio/blob/44238d033e1503c694782925d647811380a067c2/include/boost/asio/local/detail/impl/endpoint.ipp#L109-L114

And the buffer size is set here:

https://github.com/boostorg/asio/blob/44238d033e1503c694782925d647811380a067c2/include/boost/asio/detail/socket_types.hpp#L214

which in turn is derived from

https://man7.org/linux/man-pages/man7/unix.7.html

In other words, I think this is ultimately set by the kernel and so there's not much we can do -- this is a hard limit on the path size. An alternative resolution would be for us to create the RPC socket in an alternate location?

See also: https://github.com/rstudio/rstudio/blob/f46af4b9c6c03fa458ea18159fb1dcf38da1c962/src/cpp/server/include/server/ServerPaths.hpp#L35-L39
